src/EA.Weee.RequestHandlers.Tests.Unit/Admin/AatfReports/GetAllAatfSentOnDataCsvHandlerTests.cs

```diff
@@ -0,0 +1,202 @@
+            Func<Task> action = async () => await handler.HandleAsync(request);
+
+            // Assert
+            await Assert.ThrowsAsync<ArgumentException>(action);
+        }
+
+        [Theory]
+        [InlineData(2019, "", "", null, null)]
+        [InlineData(2019, "B2C", "", null, null)]
+        [InlineData(2019, "", "A", null, null)]
+        public async Task GetAllAatfSentOnDataCsvHandler_VariousParameters_ReturnsFileContent(int complianceYear, 
+           string obligationType, string aatfName, Guid? authority, Guid? panArea)
+        {
+            var authorization = new AuthorizationBuilder().AllowInternalAreaAccess().Build();
+            var context = A.Fake<WeeeContext>();
+            var commanDataAccess = A.Fake<ICommonDataAccess>();
+            var csvWriterFactory = A.Fake<CsvWriterFactory>();
+            var storedProcedures = A.Fake<IStoredProcedures>();
+
+            A.CallTo(() => context.StoredProcedures)
+                .Returns(storedProcedures);
+
+            DataSet sentOnDataSet = CreateDummyDataSet();
+
+            A.CallTo(() => storedProcedures
+            .GetAllAatfSentOnDataCsv(complianceYear, aatfName, obligationType, authority, panArea))
+            .Returns(sentOnDataSet);
+
+            var handler = new GetAllAatfSentOnDataCsvHandler(authorization, context, csvWriterFactory, commanDataAccess);
+            var request = new GetAllAatfSentOnDataCsv(complianceYear, obligationType, aatfName, authority, panArea);
+
+            // Act
+            CSVFileData data = await handler.HandleAsync(request);
+
+            // Assert
+            Assert.NotEmpty(data.FileContent);
+        }
+
+        [Fact]
+        public async Task GetAllAatfSentOnDataCsvHandler_MandatoryParameters_ReturnsFileName()
+        {
+            var authorization = new AuthorizationBuilder().AllowInternalAreaAccess().Build();
+            var context = A.Fake<WeeeContext>();
+            var commanDataAccess = A.Fake<ICommonDataAccess>();
+            var csvWriterFactory = A.Fake<CsvWriterFactory>();
+            int complianceYear = 2019;
+
+            var handler = new GetAllAatfSentOnDataCsvHandler(authorization, context, csvWriterFactory, commanDataAccess);
+            var request = new GetAllAatfSentOnDataCsv(complianceYear, string.Empty, string.Empty, null, null);
+
+            // Act
+            CSVFileData data = await handler.HandleAsync(request);
+
+            // Assert
+            Assert.Contains("2019", data.FileName);
+        }
+
+        [Fact]
+        public async Task GetAllAatfSentOnDataCsvHandler_StringParameters_ReturnsFileName()
+        {
+            var authorization = new AuthorizationBuilder().AllowInternalAreaAccess().Build();
+            var context = A.Fake<WeeeContext>();
+            var commanDataAccess = A.Fake<ICommonDataAccess>();
+            var csvWriterFactory = A.Fake<CsvWriterFactory>();
+            int complianceYear = 2019;
+            string obligationType = "B2C";
```

Comment:
Can we add test for when some of the parameters are null, probably easier to split it out into test for parameter combinations

Reply:
Added